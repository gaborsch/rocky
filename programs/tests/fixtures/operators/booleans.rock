X is true
Y is false
say X           (true)
say X and Y     (false)
say X or Y      (true)
say not X       (false)
say not X and not Y (false)
say not X or not Y (true)
say not X and not Y and not X or not Y (true)


shout 0 or 42
shout 0 or 0
shout 16 and 42
shout 16 and 0
shout 16 or 42
shout 16 or 0
shout 0 and 42
shout 0 and 0


