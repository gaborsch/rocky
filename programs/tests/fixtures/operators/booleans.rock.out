true
false
true
false
false
true
true
-----------
42
0
16
16
0
0
42
0
-----------
true
false
false
false
true
true
true
false
